openmeteo_requests/Client.py

```diff
@@ -115,18 +119,16 @@ async def _request(
     ) -> list[WeatherApiResponse]:
         params["format"] = _FLAT_BUFFERS_FORMAT
 
-        response: niquests.Response
-        async with self._session as sess:
-            method = method.upper()
-            if method == HTTPVerb.GET:
-                meth = partial(
-                    sess.get, url, params=params, verify=verify, **kwargs
-                )
-            if method == HTTPVerb.POST:
-                meth = partial(
-                    sess.post, url, data=params, verify=verify, **kwargs
-                )
-            response = await meth()
+        method = method.upper()
+        if method == HTTPVerb.GET:
+            meth = partial(
+                niquests.aget, url, params=params, verify=verify, **kwargs
+            )
+        if method == HTTPVerb.POST:
+            meth = partial(
+                niquests.apost, url, data=params, verify=verify, **kwargs
```

Comment:
@patrick-zippenfenig , also, please confirm how the following logic from the [v1.5.0...v1.6.0](https://github.com/open-meteo/python-requests/compare/v1.5.0...v1.6.0) meets the tests:

```python
        if method.upper() == "POST":
            response = self.session.request("POST", url, data=params, verify=verify, **kwargs)
        else:
            response = self.session.request("GET", url, params=params, verify=verify, **kwargs)
```
Does `self._session.request` save the situation or maybe the tests need improvements?